admin/auth_code.go

```diff
@@ -0,0 +1,50 @@
+package admin
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"time"
+
+	"github.com/rilldata/rill/admin/database"
+)
+
+const TokenExpirationSeconds = 60 * 10 // 10 minutes =
+
+func (s *Service) IssueAuthCode(ctx context.Context, clientID string) (*database.AuthCode, error) {
+	authCode, err := generateDeviceAndUserCode()
+	if err != nil {
+		return nil, err
+	}
+	authCode.ClientID = clientID
+	err = s.DB.CreateAuthCode(ctx, authCode)
+	if err != nil {
+		return nil, err
+	}
+	return authCode, nil
+}
+
+// generateDeviceAndUserCode generates a random device code and user code.
+func generateDeviceAndUserCode() (*database.AuthCode, error) {
+	// Generate a random 24-byte device code, after base64 encoding it will be 32 characters
+	deviceCodeBytes := make([]byte, 24)
+	_, err := rand.Read(deviceCodeBytes)
+	if err != nil {
+		return nil, err
+	}
+	deviceCode := base64.StdEncoding.EncodeToString(deviceCodeBytes)
+
+	// Generate an 8-character base 36 user code from the device code
+	base36Chars := "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	userCode := ""
+	for i := 0; i < 8; i++ {
+		b := deviceCodeBytes[i*3]
+		userCode += base36Chars[b%36 : b%36+1]
+	}
```

Comment:
could not find any recommended way so I came up with this, let me know if you have any better idea

Reply:
In that case, I think it might be better to randomly generate it using `crypto/rand`. I think you can use an approach like this (with n = 36) to get a base36 string: https://github.com/rilldata/rill-developer/blob/1f77017a5ed3f993635e8b6b75012a5b202af44d/admin/pkg/authtoken/authtoken.go#L114